server/src/lib.rs

```diff
@@ -970,6 +982,24 @@ async fn get_store_bytes(
     Ok(b)
 }
 
+// get the bytes for the database rule config file, if it exists,
+// otherwise it returns none.
+async fn get_database_config_bytes(
+    location: &object_store::path::Path,
+    store: &ObjectStore,
+) -> Result<Option<bytes::BytesMut>> {
```

Comment:
I wonder if the error handling would be simpler if this always returned an Error

```suggestion
) -> Result<bytes::BytesMut> {
```

And the "error is that no database config is present" was encoded as one of the error types.

this way is fine too, I was just thinking

Reply:
Yes makes sense. The retry loop would match against that kind of error, warn and return.

I can do it in Monday.

Reply:
done